### ISSUE TITLE
sendrecv.py core dump

### ISSUE THREAD

**Issue by mohammadbhuyan** [centricular/gstwebrtc-demos#44]
I am trying to build my own python app based on the demo example. After facing a core dump in application code, I ended up trying out the demo project and generated the crash.

1. I have an Ubuntu VM in VirtualBox on Mac

```
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.1 LTS
Release:	18.04
Codename:	bionic

>gst-launch-1.0 --version
gst-launch-1.0 version 1.14.1
GStreamer 1.14.1
https://launchpad.net/distros/ubuntu/+source/gstreamer1.0
```

2. In Chrome I go to https://webrtc.nirbheek.in/ and take note of the ID

3. I run sendrecv and generate the following crash. When the app starts running I see the web app showing the connection attempt. 

```
$ python webrtc-sendrecv.py 1128
Sending offer:
v=0
o=- 636836448770024085 0 IN IP4 0.0.0.0
s=-
t=0 0
a=ice-options:trickle
m=video 9 UDP/TLS/RTP/SAVPF 97
c=IN IP4 0.0.0.0
a=setup:actpass
a=ice-ufrag:1iR+VFzOHqq+nocQ22Gd1xqdzUs0bSd6
a=ice-pwd:HE4bf35FGYjB3ED54jAK/kY4LCKe//pv
a=sendrecv
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:97 VP8/90000
a=rtcp-fb:97 nack
a=rtcp-fb:97 nack pli
a=mid:video0
a=fingerprint:sha-256 89:CB:65:3D:23:38:BF:23:F1:D7:58:FC:0C:3B:14:FF:8A:F5:08:C0:50:56:1C:53:2C:10:96:33:49:9C:76:76
m=audio 9 UDP/TLS/RTP/SAVPF 96
c=IN IP4 0.0.0.0
a=setup:actpass
a=ice-ufrag:CD40Yis0r6HFzinOAcsU6B4MVMj4tmGB
a=ice-pwd:Olvo42cRw21hoGxVfjxKA26JCn5MMQe8
a=sendrecv
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:96 OPUS/48000/2
a=rtcp-fb:96 nack
a=rtcp-fb:96 nack pli
a=mid:audio1
a=fingerprint:sha-256 89:CB:65:3D:23:38:BF:23:F1:D7:58:FC:0C:3B:14:FF:8A:F5:08:C0:50:56:1C:53:2C:10:96:33:49:9C:76:76


** (python:15882): WARNING **: 00:00:42.698: (gstdtlsconnection.c:526):gst_dtls_connection_process: runtime check failed: (!priv->bio_buffer)

** (python:15882): WARNING **: 00:00:42.796: (gstdtlsconnection.c:526):gst_dtls_connection_process: runtime check failed: (!priv->bio_buffer)
Received answer:
v=0
o=- 5083558707956874919 2 IN IP4 127.0.0.1
s=-
t=0 0
a=msid-semantic: WMS 7xGKkNmER2WbqKg7BXGu2QqgwsESyM1PTfNQ
m=video 45472 UDP/TLS/RTP/SAVPF 97
c=IN IP4 10.0.2.15
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:547260449 1 udp 2122260223 10.0.2.15 45472 typ host generation 0 network-id 1
a=ice-ufrag:UsdR
a=ice-pwd:9qFkfTaQRoK8neXw0ZmUeKvP
a=ice-options:trickle
a=fingerprint:sha-256 FD:6A:07:A1:4A:5C:F2:1B:D9:40:38:84:68:64:E6:A0:2E:FB:CF:5C:98:68:7A:3D:E0:A2:4C:F6:C6:0F:9C:00
a=setup:active
a=mid:video0
a=sendrecv
a=rtcp-mux
a=rtcp-rsize
a=rtpmap:97 VP8/90000
a=rtcp-fb:97 nack
a=rtcp-fb:97 nack pli
a=ssrc:1152445112 cname:VL7j9bZj7Dl9YCIS
a=ssrc:1152445112 msid:7xGKkNmER2WbqKg7BXGu2QqgwsESyM1PTfNQ 6f4236c8-d498-4bbd-9631-49ce6b8a4826
a=ssrc:1152445112 mslabel:7xGKkNmER2WbqKg7BXGu2QqgwsESyM1PTfNQ
a=ssrc:1152445112 label:6f4236c8-d498-4bbd-9631-49ce6b8a4826
m=audio 56874 UDP/TLS/RTP/SAVPF 96
c=IN IP4 10.0.2.15
a=rtcp:9 IN IP4 0.0.0.0
a=candidate:547260449 1 udp 2122260223 10.0.2.15 56874 typ host generation 0 network-id 1
a=ice-ufrag:PVkF
a=ice-pwd:cJ/QcFwq7cI2fNbQ7X34QOmH
a=ice-options:trickle
a=fingerprint:sha-256 FD:6A:07:A1:4A:5C:F2:1B:D9:40:38:84:68:64:E6:A0:2E:FB:CF:5C:98:68:7A:3D:E0:A2:4C:F6:C6:0F:9C:00
a=setup:active
a=mid:audio1
a=sendrecv
a=rtcp-mux
a=rtpmap:96 OPUS/48000/2
a=fmtp:96 minptime=10;useinbandfec=1
a=ssrc:2421484089 cname:VL7j9bZj7Dl9YCIS
a=ssrc:2421484089 msid:7xGKkNmER2WbqKg7BXGu2QqgwsESyM1PTfNQ c2fd5046-541b-47f1-9b5c-c98d59d56d58
a=ssrc:2421484089 mslabel:7xGKkNmER2WbqKg7BXGu2QqgwsESyM1PTfNQ
a=ssrc:2421484089 label:c2fd5046-541b-47f1-9b5c-c98d59d56d58

double free or corruption (out)
Aborted (core dumped)
```

I am looking for ways to genberate some debug info so that I have more clue.

**Comment by lpasarel** [comment]
May be related to gstreamer dtlssrtpenc/dtlssrtpdec. Do you have it properly installed?

Can you post  the output of `gst-inspect-1.0 dtlssrtpenc`and `gst-inspect-1.0 dtlssrtpdec`?

Also, you can try to launch  a local [signaling python server](../tree/master/signalling/simple-server.py) without SSL `./simple-server.py --disable-ssl` and check if same error

**Comment by lpasarel** [comment]
Also, you need at least  gstreamer 1.14.2 for running python examples as specified in [Readme#examples](../tree/master/README.md#examples)

> The python version requires at least version 1.14.2 of gstreamer and its plugins.

**Comment by mohammadbhuyan** [comment]
I have managed to build GStreamer 1.14.2 and its plugins on Ubuntu 18.04.

I have the C version of the demo running (using https://webrtc.nirbheek.in/ for signalling)

But the python version fails. The error is already reported in:

https://github.com/centricular/gstwebrtc-demos/issues/42

So I will provide some details there.

**Comment by mohammadbhuyan** [comment]
This can be closed as I have progressed to get the python version going based on the fix provided in #42 but facing another error. I will open a new issue for that.
